presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -561,6 +566,120 @@ private void invokeNoMoreSplitsIfNecessary()
+            if (principal.isPresent()) {
+                targetJob.set(MRConfig.FRAMEWORK_NAME, MRConfig.CLASSIC_FRAMEWORK_NAME);
+                targetJob.set(MRConfig.MASTER_USER_NAME, principal.get().getName());
+            }
+            if (fileInputFormat instanceof TextInputFormat) {
+                ((TextInputFormat) fileInputFormat).configure(targetJob);
+            }
+            FileInputFormat.setInputPaths(targetJob, targetPath);
+            InputSplit[] targetSplits = hdfsEnvironment.doAs(
+                    hdfsContext.getIdentity().getUser(),
+                    () -> fileInputFormat.getSplits(targetJob, 0));
+
+            InternalHiveSplitFactory splitFactory = new InternalHiveSplitFactory(
+                    targetFilesystem,
+                    partitionName,
+                    inputFormat,
+                    schema,
+                    partitionKeys,
+                    effectivePredicate,
+                    partitionMatchSupplier,
+                    partition.getTableToPartitionMapping(),
+                    Optional.empty(),
+                    getMaxInitialSplitSize(session),
+                    isForceLocalScheduling(session),
+                    s3SelectPushdownEnabled);
+            lastResult = addSplitsToSource(targetSplits, splitFactory);
+            if (stopped) {
+                return COMPLETED_FUTURE;
+            }
+        }
+        return lastResult;
+    }
+
+    private Iterator<InternalHiveSplit> buildManifestFileIterator(Table table,
+            Path parent,
+            List<Path> paths,
+            DirectoryLister directoryLister,
+            NamenodeStats namenodeStats,
+            InternalHiveSplitFactory splitFactory,
+            boolean splittable)
+            throws DirectoryNotAllowedException
+    {
+        Map<Path, LocatedFileStatus> fileStatuses = new HashMap();
+        try {
+            FileSystem targetFilesystem = hdfsEnvironment.getFileSystem(hdfsContext, parent);
+            HiveFileIterator fileStatusIterator = new HiveFileIterator(table, parent, targetFilesystem, directoryLister, namenodeStats, IGNORED, true);
+            fileStatusIterator.forEachRemaining(status -> fileStatuses.putIfAbsent(status.getPath(), status));
+        }
+
+        catch (IOException e) {
+            throw new PrestoException(HIVE_FILE_NOT_FOUND, String.format("No Filesystem available for %s", parent), e);
+        }
+
+        List<LocatedFileStatus> locatedFileStatuses = paths.stream().map(this::canonicalizeSymlinkPath).map(path -> fileStatuses.get(path)).collect(Collectors.toList());
+        if (locatedFileStatuses.contains(null)) {
+            // This check will catch all directories in the manifest since HiveFileIterator will not return any directories
+            throw new DirectoryNotAllowedException();
+        }
+
+        return hdfsEnvironment.doAs(hdfsContext.getIdentity().getUser(), () -> locatedFileStatuses.stream()
```

Comment:
I don't think you need `doAs` as `createInternalHiveSplitIterator` is used directly from `loadPartition`.